rita_common/src/peer_listener/mod.rs

```diff
@@ -117,8 +119,11 @@ pub fn peerlistener_tick() {
 
     let (a, b) = receive_im_here(&mut writer.interfaces);
     {
-        (*writer).peers = a;
-        (*writer).interface_map = b;
+        // Reset hashmaps only when not contacting peers
+        if !writer.contacting_neighbors {
```

Comment:
I'm not sure this will work how you expect, what if for some reason we're stuck contacting neighbors constantly? Lets try this out on an actual router first.

Reply:
I changed it to blocking the entire peer listener tick. This emulates what we were doing before when we held the lock out the for loop that calls contact_neighbors(), The only time we are stuck contacting neighbors is on a gateway when we run to_socket_addrs(), but as long as it times out, we run the tick